webapp/src/pages/Settings.tsx

```diff
@@ -2,58 +2,618 @@ import {
         setPartialConfig({ ...partialConfig, [field]: checked ? {} : null })
       }
     />
   );
 
+  const displayReadonlyFields = (label: string, value: string | undefined) => (
+    <TextField
+      key={String(value)}
+      sx={{
+        m: 1,
+        width: "32ch",
+      }}
+      size="small"
+      variant="standard"
+      label={<Typography fontWeight="bold">{label}</Typography>}
+      value={value}
+      disabled={isError || isFetching}
+      InputProps={{
+        readOnly: true,
+        disableUnderline: true,
+        style: { textTransform: "capitalize", fontSize: 14 },
+      }}
+    />
+  );
+
+  const displaySubFields = (
+    config: keyof AzimuthConfig,
+    field: string,
+    value: number | undefined
+  ) => (
+    <TextField
+      key={field}
+      sx={{ m: 1, width: "16ch" }}
+      size="small"
+      label={field}
+      type="number"
+      defaultValue={value}
+      disabled={!Boolean(resultingConfig[config])}
+      inputProps={STEPPER[field]}
+      InputProps={{ style: { fontSize: 14 } }}
+      variant="standard"
+      onChange={(event) =>
+        setPartialConfig(
+          _.merge(partialConfig, {
+            [config]: { [field]: Number(event.target.value) },
+          })
+        )
+      }
+    />
+  );
+
+  const displayPostprocessorSubField = (
+    pipeline: PipelineDefinition,
+    field: string,
+    postprocessorIdx: number,
+    value: number | undefined
+  ) => (
+    <TextField
+      key={field}
+      sx={{ m: 1, width: "16ch" }}
+      size="small"
+      label={field}
+      type="number"
+      InputLabelProps={{
+        shrink: true,
+      }}
+      defaultValue={value}
+      inputProps={{ min: 0, max: 1, step: 0.1 }}
+      variant="standard"
+      onChange={(event) => {
+        const existedPipeline = partialConfig.pipelines
+          ? partialConfig.pipelines.find(({ name }) => name === pipeline.name)
+          : undefined;
+        const current_postprocessor = pipeline?.postprocessors?.find(
+          (_, index) => index === postprocessorIdx
+        );
+        const updatedPostprocessor = _.assign({}, current_postprocessor, {
+          [field]: Number(event.target.value),
+        });
+        const updatedPipeline = [
+          {
+            ...pipeline,
+            postprocessors: _.unionBy(
+              [updatedPostprocessor],
+              pipeline.postprocessors,
+              "class_name"
+            ),
+          },
+        ];
+        existedPipeline
+          ? setPartialConfig({
+              ...partialConfig,
+              pipelines: _.unionBy(
+                updatedPipeline,
+                partialConfig.pipelines,
+                "name"
+              ),
+            })
+          : setPartialConfig({
+              ...partialConfig,
+              pipelines: _.unionBy(
+                updatedPipeline,
+                resultingConfig?.pipelines,
+                "name"
+              ),
+            });
```

Comment:
I think this is equivalent to this simpler form:
```suggestion
        setPartialConfig({
          ...partialConfig,
          pipelines: _.unionBy(
            updatedPipeline,
            existedPipeline ? partialConfig.pipelines : resultingConfig.pipelines,
            "name"
          ),
        });
```

Reply:
By the way, sorry for repeating, but you can remove the 19 unnecessary `?`s after `resultingConfig`, since it is always defined.

Reply:
Second reading... It seems overkill to search for the pipeline with `_.unionBy(..., "name")`. Why not pass the pipeline index to `displayPostprocessorSubField`?